### PR TITLE
[online-cmaq] Fix/canopy fengsha config

### PR DESCRIPTION
**Pull request by zmoon**
## DESCRIPTION OF CHANGES: 

* Add `fengsha` to the `emission_sources` list in `aqm.rc`.
  - The Jinja syntax used was demoed [here](https://github.com/noaa-oar-arl/ufs-srweather-app/pull/1#discussion_r1013028559).
* Update the `ARL_run` sample configs
  - add Patrick's WCOSS2 config
  - set `DO_AQM_CANOPY: false` in all and add the `DO_AQM_` settings to the WCOSS2 config

cc: @bbakernoaa @drnimbusrain

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## CHECKLIST

- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

### CODE REVIEW

**Comment by chan-hoo** [comment]
@zmoon , please add [online-cmaq] to the title like `[online-cmaq] fix canopy fengsha config` to avoid any confusion by other branch reviewers.

File: parm/aqm.rc [hunk 1]
```diff
@@ -29,11 +29,9 @@ ctm_pmdiag: true
 #
 #  Input emissions
 #
-{%- if do_aqm_canopy %}
-emission_sources: anthro fire canopy
-{%- else %}
 emission_sources: anthro fire
-{%- endif %}
+{%- if do_aqm_canopy %} canopy{% endif %}
+{%- if do_aqm_fengsha %} fengsha{% endif %}
```
Comment (chan-hoo) [inline parm/aqm.rc:35]: @zmoon, with this `fengsha` flag, the `run_fcst' task fails. I think the line 34 should be removed. If you succeeded with this flag, there should be any other changes somewhere else. Please check this again.

Comment (bbakernoaa) [reply parm/aqm.rc:35]: @chan-hoo I believe that this is a different issue as 1) without this fengsha is not actually being run as `fengsha` is not included as an emission source.  2) the file being read in within the sample config has the wrong dimensions and will crash.  I'm running a test right now with the updated file that should fix this issue. I'll create a separate issue for this once I have a day completed to show.

Comment (chan-hoo) [reply parm/aqm.rc:35]: @bbakernoaa, if so, you can put your change into this PR. Or, remove Ln 34 from this PR and put it into your new PR. In any cases, this `fengsha` flag should come with the valid configuration.

Comment (bbakernoaa) [reply parm/aqm.rc:35]: @chan-hoo I completely agree I'll link the issue (which is just the input file) to this PR and get it put into the sample configuration files

Comment (zmoon) [reply parm/aqm.rc:35]: On a related note, the EMC sample configs have `DO_AQM_FENGSHA: true` but we have false. Do we want things to stay that way?

Comment (chan-hoo) [reply parm/aqm.rc:35]: @zmoon, good point! I am not sure. Based on your comments, I think the `fengsha` option does not work correctly in the current status of the workflow. Is this correct? @JianpingHuang-NOAA, are you aware of this issue?

Comment (zmoon) [reply parm/aqm.rc:35]: @chan-hoo Barry said that with the fixed input file it works, we will get that fix in here soon.

Comment (zmoon) [reply parm/aqm.rc:35]: @chan-hoo I went ahead and updated the path in the sample configs based on Barry's note in https://github.com/NOAA-EMC/AQM/issues/36

Comment (chan-hoo) [reply parm/aqm.rc:35]: @zmoon, got it. Thanks. I'll test it soon.

Comment (chan-hoo) [reply parm/aqm.rc:35]: it looks working well.

File: ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2 [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  RUN_ADD_AQM_CHEM_LBCS: true
+  RUN_ADD_AQM_GEFS_LBCS: true
+  AQM_GEFS_CYC: 0
+  AQM_RC_TMPL_FN: aqm.rc
+  AQM_CONFIG_DIR: /lfs/h2/emc/lam/noscrub/RRFS_CMAQ/aqm/epa/data
+  AQM_BIO_DIR: /lfs/h2/emc/lam/noscrub/RRFS_CMAQ/aqm/bio
+  AQM_BIO_FILE: BEIS_RRFScmaq_C775.ncf
+  DO_AQM_FENGSHA: false
+  AQM_FENGSHA_DIR: /lfs/h2/emc/ptmp/barry.baker
+  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs_C793_AQM_13km
```
Comment (chan-hoo) [inline ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2]: @zmoon @bbakernoaa , as can be seen in ush/create_aqm_rc_file.py, the format of the input file is "    aqm_fengsha_file_fn=AQM_FENGSHA_FILE_PREFIX+"_"+PREDEF_GRID_NAME+AQM_FENGSHA_FILE_SUFFIX". Have you tested these sample scripts before commit? If you turn Fengsha on, this script will fail. The name of the input file should be "FENGSHA_p8_10km_inputs_AQM_NA_13km.nc" and the above parameter "AQM_FENGSHA_FILE_PREFIX" should be "FENGSHA_p8_10km_inputs".

File: ush/sample_config/ARL_run/config.yaml_community_aqmna13km_1day_hera [hunk 3]
```diff
@@ -73,10 +73,10 @@ cpl_aqm_parm:
   AQM_BIO_DIR: /scratch2/NCEPDEV/naqfc/RRFS_CMAQ/aqm/bio
   AQM_BIO_FILE: BEIS_RRFScmaq_C775.ncf
   DO_AQM_FENGSHA: false
-  AQM_FENGSHA_DIR: /scratch2/NCEPDEV/naqfc/RRFS_CMAQ/FENGSHA
-  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs
+  AQM_FENGSHA_DIR: /scratch1/RDARCH/rda-arl-gpu/Barry.Baker/emissions/nexus/FENGSHA/Online_CMAQ
+  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs_C793_AQM_13km
   AQM_FENGSHA_FILE_SUFFIX: .nc
```
Comment (chan-hoo) [inline ush/sample_config/ARL_run/config.yaml_community_aqmna13km_1day_hera:78]: Same as above. "_C793_AQM_13km" should be removed.

File: ush/sample_config/ARL_run/config.yaml_community_conus13km_1day_hera [hunk 4]
```diff
@@ -62,10 +62,10 @@ cpl_aqm_parm:
   AQM_BIO_DIR: /scratch2/NCEPDEV/naqfc/RRFS_CMAQ/aqm/bio
   AQM_BIO_FILE: BEIS_RRFScmaq_C775.ncf
   DO_AQM_FENGSHA: false
-  AQM_FENGSHA_DIR: /scratch2/NCEPDEV/naqfc/RRFS_CMAQ/FENGSHA
-  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs
+  AQM_FENGSHA_DIR: /scratch1/RDARCH/rda-arl-gpu/Barry.Baker/emissions/nexus/FENGSHA/Online_CMAQ
+  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs_C793_AQM_13km
```
Comment (chan-hoo) [inline ush/sample_config/ARL_run/config.yaml_community_conus13km_1day_hera]: Same as above. "_C793_AQM_13km" should be removed.

File: ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_hera [hunk 5]
```diff
@@ -75,8 +75,8 @@ cpl_aqm_parm:
   AQM_BIO_DIR: /scratch2/NCEPDEV/naqfc/RRFS_CMAQ/aqm/bio
   AQM_BIO_FILE: BEIS_RRFScmaq_C775.ncf
   DO_AQM_FENGSHA: true
-  AQM_FENGSHA_DIR: /scratch2/NCEPDEV/naqfc/RRFS_CMAQ/FENGSHA
-  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs
+  AQM_FENGSHA_DIR: /scratch1/RDARCH/rda-arl-gpu/Barry.Baker/emissions/nexus/FENGSHA/Online_CMAQ
+  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs_C793_AQM_13km
   AQM_FENGSHA_FILE_SUFFIX: .nc
```
Comment (chan-hoo) [inline ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_hera]: I have updated the input file for EMC. Please remove this change.

File: ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_wcoss2 [hunk 6]
```diff
@@ -75,8 +75,8 @@ cpl_aqm_parm:
   AQM_BIO_DIR: /lfs/h2/emc/lam/noscrub/RRFS_CMAQ/aqm/bio
   AQM_BIO_FILE: BEIS_RRFScmaq_C775.ncf
   DO_AQM_FENGSHA: true
-  AQM_FENGSHA_DIR: /lfs/h2/emc/physics/noscrub/jianping.huang/data/emissions/nexus/Dust
-  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs
+  AQM_FENGSHA_DIR: /lfs/h2/emc/ptmp/barry.baker
+  AQM_FENGSHA_FILE_PREFIX: FENGSHA_p8_10km_inputs_C793_AQM_13km
   AQM_FENGSHA_FILE_SUFFIX: .nc
```
Comment (chan-hoo) [inline ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_wcoss2]: I have updated the input file for EMC. Please remove this change.

**Comment by chan-hoo** [comment]
@zmoon, please update your branch with the latest one to resolve conflicts.

**Comment by bbakernoaa** [comment]
@chan-hoo I think that we are making this too difficult with all the applications and combinations to this input file.  Why can we not simply list the entire path to the file instead of 3 separate variables.  This seems like over engineering for this simple task.

**Comment by chan-hoo** [comment]
@bbakernoaa, that is because we don't want to make change for each domain. The domain name is automatically put in the file name. So, we don't have to worry about the domain change. For the suffix, many developers create their own version in many cases such as "_v1.0.nc" and "_modified.nc". This suffix parameter allows this kind of flexibility.

File: ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2 (hunk 2 above)
Comment (zmoon) [reply ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2]: Sorry about that, didn't understand that the domain was added automatically.  Will fix.  I imagine Barry tested with a direct aqm.rc, not with aqm.rc creation from template.

Comment (zmoon) [reply ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2]: Resolved in 2b85611

File: ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_hera (hunk 5 above)
Comment (zmoon) [reply ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_hera]: Reverted my change in 673dc1e

**Review by chan-hoo — APPROVED** [review]

File: parm/aqm.rc (hunk 1 above)
Comment (zmoon) [reply parm/aqm.rc:35]: > On a related note, the EMC sample configs have `DO_AQM_FENGSHA: true` but we have false. Do we want things to stay that way?

@bbakernoaa do you want to change this in our sample configs moving forward? or leave it this way?

Comment (bbakernoaa) [reply parm/aqm.rc:35]: We should definately add this to the arl sample config.  We are the the developers of this.....

**Comment by bbakernoaa** [comment]
@chan-hoo thanks

File: parm/aqm.rc (hunk 1 above)
Comment (zmoon) [reply parm/aqm.rc:35]: Ok, I can add that to our branch when I update it today. Do we want to also do a PR for that to here?

**Pull request merged**